src/Algolia.Search/Models/Rules/FacetOrdering.cs

```diff
@@ -0,0 +1,43 @@
+/*
+* Copyright (c) 2021 Algolia
+* http://www.algolia.com/
+*
+* Permission is hereby granted, free of charge, to any person obtaining a copy
+* of this software and associated documentation files (the "Software"), to deal
+* in the Software without restriction, including without limitation the rights
+* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+* copies of the Software, and to permit persons to whom the Software is
+* furnished to do so, subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be included in
+* all copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+* THE SOFTWARE.
+*/
+
+using System.Collections.Generic;
+
+namespace Algolia.Search.Models.Rules
+{
+    /// <summary>
+    /// Facets and facets values ordering rules container.
+    /// </summary>
+    public class FacetOrdering
+    {
+        /// <summary>
+        /// Define or override the way facet attributes are displayed.
+        /// </summary>
+        public FacetsOrder Facets { get; set; }
```

Comment:
these are optional, how is that defined in c#?

Reply:
If I understood correctly, reference types such as class instances are nullable (optional) by definition, so no need to declare them as optional explicitly. Could you confirm @Ant-hem ?

Reply:
That's right @VladislavFitz, not present = null = not serialized.